pkg/controller/component/lifecycle/lfa_common.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package lifecycle
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	"github.com/apecloud/kubeblocks/pkg/constant"
+	"github.com/apecloud/kubeblocks/pkg/controller/component"
+)
+
+const (
+	podNameList       = "podNameList"
+	podIPList         = "podIPList"
+	podHostNameList   = "podHostNameList"
+	podHostIPList     = "podHostIPList"
+	compName          = "compName"
+	compReplicas      = "compReplicas"
+	deletingCompList  = "deletingCompList"
+	undeletedCompList = "undeletedCompList"
+)
+
+func getClusterPods(ctx context.Context, cli client.Reader, namespace, clusterName string) (map[string]string, error) {
+	// TODO
+	pods, err := component.ListOwnedPods(ctx, cli, namespace, clusterName, "")
+	if err != nil {
+		return nil, err
+	}
+	name := func(pod *corev1.Pod) string {
+		return pod.Name
+	}
+	IP := func(pod *corev1.Pod) string {
+		return pod.Status.PodIP
+	}
+	hostName := func(pod *corev1.Pod) string {
+		return pod.Spec.Hostname
```

Comment:
Do we need this parameter? it's equal to Pod's name in KB.

according to
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields

The Pod spec has an optional hostname field, which can be used to specify a different hostname. When specified, it takes precedence over the Pod's name to be the hostname of the Pod (again, as observed from within the Pod). For example, given a Pod with spec.hostname set to "my-host", the Pod will have its hostname set to "my-host".

Reply:
InstanceSet controller set pod.spec.hostname to pod's name

```
	pod := builder.NewPodBuilder(parent.Namespace, name).
		AddAnnotationsInMap(template.Annotations).
		AddLabelsInMap(template.Labels).
		AddLabelsInMap(labels).
		AddControllerRevisionHashLabel(revision).
		AddLabelsInMap(map[string]string{constant.KBAppPodNameLabelKey: name}).
		SetPodSpec(*template.Spec.DeepCopy()).
		GetObject()
	// Set these immutable fields only on initial Pod creation, not updates.
	pod.Spec.Hostname = pod.Name
	pod.Spec.Subdomain = getHeadlessSvcName(parent.Name)
```

Reply:
> Do we need this parameter? it's equal to Pod's name in KB.
> 
> according to https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
> 
> The Pod spec has an optional hostname field, which can be used to specify a different hostname. When specified, it takes precedence over the Pod's name to be the hostname of the Pod (again, as observed from within the Pod). For example, given a Pod with spec.hostname set to "my-host", the Pod will have its hostname set to "my-host".

here should be NodeName，not HostName？